include/rmm/mr/device/fixed_multisize_memory_resource.hpp

```diff
@@ -139,7 +139,7 @@ class fixed_multisize_memory_resource : public device_memory_resource {
    *
    * @return UpstreamResource* the upstream memory resource.
    */
-  Upstream* get_upstream() const noexcept { return upstream_mr_; }
+  Upstream_ptr get_upstream() const noexcept { return upstream_mr_; }
```

Comment:
This won't work if `Upstream_ptr` is `unique_ptr`.

Reply:
Fixed.